### ISSUE TITLE
`AutoModel.from_pretrained` sends config kwargs to model

### ISSUE THREAD

**Issue by bryant1410** [huggingface/transformers#2985]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Bert (may apply to more)

Language I am using the model on (English, Chinese ...): English

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

```python
from transformers import AutoModel
AutoModel.from_pretrained('bert-base-uncased', output_attention=True)
```

([example from the docs](https://huggingface.co/transformers/model_doc/auto.html#transformers.AutoModel.from_pretrained))

It crashes:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "[...]/transformers/src/transformers/modeling_auto.py", line 384, in from_pretrained
    return model_class.from_pretrained(pretrained_model_name_or_path, *model_args, config=config, **kwargs)
  File "[...]/transformers/src/transformers/modeling_utils.py", line 463, in from_pretrained
    model = cls(config, *model_args, **model_kwargs)
TypeError: __init__() got an unexpected keyword argument 'output_attention'
```

## Expected behavior

That the code returns a correct model, without crashing

## Environment info

     
- `transformers` version: 2.5.0 (master)
- Platform: Linux-4.15.0-76-generic-x86_64-with-debian-buster-sid
- Python version: 3.7.4
- PyTorch version (GPU?): 1.4.0 (True)
- Tensorflow version (GPU?): 2.0.0 (False)
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: no

**Comment by bryant1410** [comment]
Seems related to #2694

**Comment by LysandreJik** [comment]
Indeed! I fixed it with #2998.
